metadata-ingestion/src/datahub/ingestion/source/cassandra/cassandra_profiling.py

```diff
@@ -0,0 +1,240 @@
+        profile_aspect = self.populate_profile_aspect(profile_data)
+
+        if profile_aspect:
+            self.report.report_entity_profiled(table_name)
+            mcp = MetadataChangeProposalWrapper(
+                entityUrn=dataset_urn, aspect=profile_aspect
+            )
+            yield mcp.as_workunit()
+
+    def populate_profile_aspect(self, profile_data: ProfileData) -> DatasetProfileClass:
+        field_profiles = [
+            self._create_field_profile(column_name, column_metrics)
+            for column_name, column_metrics in profile_data.column_metrics.items()
+        ]
+        return DatasetProfileClass(
+            timestampMillis=round(time.time() * 1000),
+            rowCount=profile_data.row_count,
+            columnCount=profile_data.column_count,
+            fieldProfiles=field_profiles,
+        )
+
+    def _create_field_profile(
+        self, field_name: str, field_stats: ColumnMetric
+    ) -> DatasetFieldProfileClass:
+        quantiles = field_stats.quantiles
+        return DatasetFieldProfileClass(
+            fieldPath=field_name,
+            uniqueCount=field_stats.distinct_count,
+            nullCount=field_stats.null_count,
+            min=str(field_stats.min) if field_stats.min else None,
+            max=str(field_stats.max) if field_stats.max else None,
+            mean=str(field_stats.mean) if field_stats.mean else None,
+            median=str(field_stats.median) if field_stats.median else None,
+            stdev=str(field_stats.stdev) if field_stats.stdev else None,
+            quantiles=[
+                QuantileClass(quantile=str(0.25), value=str(quantiles[0])),
+                QuantileClass(quantile=str(0.75), value=str(quantiles[1])),
+            ]
+            if quantiles
+            else None,
+            sampleValues=field_stats.sample_values
+            if field_stats.sample_values
+            else None,
+        )
+
+    def profile_table(
+        self, keyspace_name: str, table_name: str, columns: List[CassandraColumn]
+    ) -> ProfileData:
+        profile_data = ProfileData()
+
+        resp = self.api.execute(
+            CassandraQueries.ROW_COUNT.format(keyspace_name, table_name)
+        )
+        if resp:
+            profile_data.row_count = resp[0].row_count
+
+        profile_data.column_count = len(columns)
+
+        if not self.config.profiling.profile_table_level_only:
+            resp = self.api.execute(
+                f'SELECT {", ".join([col.column_name for col in columns])} FROM {keyspace_name}."{table_name}"'
+            )
+            profile_data.column_metrics = self._collect_column_data(resp, columns)
+
+        return self._parse_profile_results(profile_data)
+
+    def _parse_profile_results(self, profile_data: ProfileData) -> ProfileData:
+        for _, column_metrics in profile_data.column_metrics.items():
+            if column_metrics.values:
+                self._compute_field_statistics(column_metrics)
```

Comment:
Please add try..except report warning here if this computation goes wrong for some column

Reply:
Also add exception handling at table level profile generation to ensure we do not fail profiling if any single table profiling fails.